### ISSUE TITLE
UseFloatingWatermark bug in TextBoxHelper

### ISSUE THREAD

**Issue by taori** [MahApps/MahApps.Metro#2252]
my app.xaml has the following configuration:

```xaml
<Application x:Class="ImmoCrawler.Client.App"
             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:local="clr-namespace:ImmoCrawler.Client"
             xmlns:controls="http://metro.mahapps.com/winfx/xaml/controls">
    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Controls.xaml" />
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Fonts.xaml" />
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Colors.xaml" />
<!--                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Accents/Blue.xaml" />-->
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Accents/Amber.xaml" />
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Accents/BaseDark.xaml" />
                <ResourceDictionary Source="Resources/Icons.xaml" />
				<ResourceDictionary>
					<local:AppBootstrapper x:Key="Bootstrapper"></local:AppBootstrapper>
					<Style TargetType="TextBox" BasedOn="{StaticResource MetroTextBox}">
						<Setter Property="controls:TextBoxHelper.UseFloatingWatermark" Value="True"/>
						<Setter Property="controls:TextBoxHelper.IsWaitingForData" Value="True"/>
					</Style>
				</ResourceDictionary>
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </Application.Resources>
</Application>
```

xaml setup which renders as expected:

```xaml
<TextBox  Text="{Binding Path=InsertionName, Mode=TwoWay, ValidatesOnNotifyDataErrors=True, UpdateSourceTrigger=PropertyChanged}"
                  controls:TextBoxHelper.Watermark="Feldname"
                  controls:TextBoxHelper.IsWaitingForData="True"
                  controls:TextBoxHelper.UseFloatingWatermark="True"
				  ToolTip="Feldname"
		          DockPanel.Dock="Bottom"
		          FontSize="20">
			<TextBox.InputBindings>
				<KeyBinding Key="Enter" Command="{Binding Path=InsertCommand}"></KeyBinding>
			</TextBox.InputBindings>
		</TextBox>
```

xaml which breaks watermark rendering if content is empty:

```xaml
<TextBox  Text="{Binding Path=InsertionName, Mode=TwoWay, ValidatesOnNotifyDataErrors=True, UpdateSourceTrigger=PropertyChanged}"
                  controls:TextBoxHelper.Watermark="Feldname"
				  ToolTip="Feldname"
		          DockPanel.Dock="Bottom"
		          FontSize="20">
			<TextBox.InputBindings>
				<KeyBinding Key="Enter" Command="{Binding Path=InsertCommand}"></KeyBinding>
			</TextBox.InputBindings>
		</TextBox>
```

This should be working shouldn't it?

error state. label where watermark is rendered if there is content inside the textbox does not collapse

![error](https://cloud.githubusercontent.com/assets/5545184/11611971/79c6a8b2-9be6-11e5-95c8-e1b0c57ec3d0.PNG)

explicitly set attached properties on textbox - works just fine:

![expected](https://cloud.githubusercontent.com/assets/5545184/11611970/79c5ed5a-9be6-11e5-805c-55c3a7a7a775.PNG)

**Comment by punker76** [comment]
@taori With which version does this happen? If it's not the latest I would suggest you to do this...

**Comment by taori** [comment]
- happened in 1.1.2, 
- resolved with https://www.nuget.org/packages/MahApps.Metro/1.1.3-ALPHA240. 

Now i'm getting some designer errors with that one though. Since it is fixed with 1.1.3-ALPHA240 i will close this ticket and see if the there is a designer issue open yet

Thanks you.
